### PR TITLE
move processors init out of main func

### PR DESCRIPTION
**Pull request by liubog2008**
I want to customize processors, so I move the init out of main func and create a processors factory, just like sources.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/heapster/1212)

### CODE REVIEW

**Comment by mwielgus** [comment]
Why do you need this change?

BTW, please sign the CLA.

**Comment by liubog2008** [comment]
I signed it!

**Comment by liubog2008** [comment]
I want to add some labels and make data more useful for us, so that I don't need to process data again after getting data from influxdb.

File: metrics/heapster.go [hunk 1]
```diff
@@ -54,6 +53,7 @@ var (
 	argAllowedUsers     = flag.String("allowed_users", "", "comma-separated list of allowed users")
 	argSources          flags.Uris
 	argSinks            flags.Uris
+	argProcessors       = flag.String("processors", "kubernetes", "processors for heapster")
```
Comment (mwielgus) [inline metrics/heapster.go]: Actually, lets name the default set "default".  As Heapster works only in kubernetes name "kubernetes" doesn't mean anything.

File: metrics/processors/kubeprocessors/cluster_aggregator.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package processors
+package kubeprocessors
```
Comment (mwielgus) [inline metrics/processors/kubeprocessors/cluster_aggregator.go]: default

File: metrics/processors/kubeprocessors/pod_based_enricher.go [hunk 3]
```diff
@@ -196,7 +201,19 @@ func intValue(value int64) core.MetricValue {
 	}
 }
 
-func NewPodBasedEnricher(podLister *cache.StoreToPodLister) (*PodBasedEnricher, error) {
+func NewPodBasedEnricher(url *url.URL) (*PodBasedEnricher, error) {
```
Comment (mwielgus) [inline metrics/processors/kubeprocessors/pod_based_enricher.go]: We pass pod listener for a reason. In large clusters with thousands of pods podsListeners are quite big (they contain a cache of ALL pods in the system with all metadata) and having 2 copies of them would be a big waste of memory.

Comment (liubog2008) [inline metrics/processors/kubeprocessors/pod_based_enricher.go]: Ok, It seems the biggest problem in my pr.  I'll try to create a public cache repository for getting cache store or indexer.

File: metrics/heapster.go (hunk 1 above)
Comment (liubog2008) [inline metrics/heapster.go]: OK, I'll change it... In fact, I just imitate what Sources do.

**Comment by piosz** [comment]
@liubog2008 This PR is inactive for a longer period of time. If you plan to work on this, please reopen.

**Pull request closed without merging**